packages/browser/lib/core/crypto.ts

```diff
@@ -248,21 +337,71 @@ export class CoreCrypto {
     );
   }
 
+  static #buildMetadata(
+    encryptionTimestamp: number,
+    role?: string
+  ): Uint8Array {
+    const bufferArray = [];
+
+    // Binary representation of a fixed map with 2 or 3 items, followed by the key-value pairs.
+    bufferArray.push(0x80 | (!role ? 2 : 3));
+
+    if (role) {
+      // `dr` (data role) => role_name
+      // Binary representation for a fixed string of length 2, followed by `dr`
+      bufferArray.push(0xa2);
+      bufferArray.push(...Array.from("dr").map((c) => c.charCodeAt(0)));
+
+      // Binary representation for a fixed string of role name length, followed by the role name itself.
+      bufferArray.push(0xa0 | role.length);
+      bufferArray.push(...Array.from(role).map((c) => c.charCodeAt(0)));
+    }
+
+    // "eo" (encryption origin) => 5 (Node SDK)
+    // Binary representation for a fixed string of length 2, followed by `eo`
+    bufferArray.push(0xa2);
+    bufferArray.push(...Array.from("eo").map((c) => c.charCodeAt(0)));
+
+    // Binary representation for the integer 5
+    bufferArray.push(5);
+
+    // "et" (encryption timestamp) => current time
+    // Binary representation for a fixed string of length 2, followed by `et`
+    bufferArray.push(0xa2);
+    bufferArray.push(...Array.from("et").map((c) => c.charCodeAt(0)));
+
+    // Binary representation for a 4-byte unsigned integer (uint 32), followed by the epoch time
+    bufferArray.push(0xce);
+    bufferArray.push((encryptionTimestamp >> 24) & 0xff);
+    bufferArray.push((encryptionTimestamp >> 16) & 0xff);
+    bufferArray.push((encryptionTimestamp >> 8) & 0xff);
+    bufferArray.push(encryptionTimestamp & 0xff);
+
+    return new Uint8Array(bufferArray);
+  }
+
   // Use unknown in interal methods
   async #traverseObject(
-    data: Datatypes.EncryptableObject
+    data: Datatypes.EncryptableObject,
+    role?: string
   ): Promise<Datatypes.EncrypedObject>;
 
   async #traverseObject(
-    data: Datatypes.EncryptableValue[]
+    data: Datatypes.EncryptableValue[],
+    role?: string
   ): Promise<Datatypes.EncryptedValue[]>;
 
-  async #traverseObject(data: Datatypes.EncryptableAsString): Promise<string>;
   async #traverseObject(
-    data: Datatypes.EncryptableValue
+    data: Datatypes.EncryptableAsString,
+    role?: string
+  ): Promise<string>;
+
+  async #traverseObject(
+    data: Datatypes.EncryptableValue,
+    role?: string
   ): Promise<Datatypes.EncryptedValue>;
 
-  async #traverseObject(data: unknown) {
+  async #traverseObject(data: unknown, role?: string) {
     if (Datatypes.isEncryptableAsString(data)) {
       return this.#encryptString(
```

Comment:
Need to pass the role to the `encryptString` call here.

Reply:
good catch